digitalocean/database/resource_database_kafka_config.go

```diff
@@ -0,0 +1,293 @@
+				Type:     schema.TypeBool,
+				Optional: true,
+				Computed: true,
+			},
+			"log_message_timestamp_difference_max_ms": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"log_preallocate": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Computed: true,
+			},
+			"log_retention_bytes": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"log_retention_hours": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+			"log_retention_ms": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"log_roll_jitter_ms": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"log_segment_delete_delay_ms": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+			"auto_create_topics_enable": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceDigitalOceanDatabaseKafkaConfigCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*config.CombinedConfig).GodoClient()
+	clusterID := d.Get("cluster_id").(string)
+
+	if err := updateKafkaConfig(ctx, d, client); err != nil {
+		return diag.Errorf("Error updating Kafka configuration: %s", err)
+	}
+
+	d.SetId(makeDatabaseKafkaConfigID(clusterID))
+
+	return resourceDigitalOceanDatabaseKafkaConfigRead(ctx, d, meta)
+}
+
+func resourceDigitalOceanDatabaseKafkaConfigUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*config.CombinedConfig).GodoClient()
+
+	if err := updateKafkaConfig(ctx, d, client); err != nil {
+		return diag.Errorf("Error updating Kafka configuration: %s", err)
+	}
+
+	return resourceDigitalOceanDatabaseKafkaConfigRead(ctx, d, meta)
+}
+
+func updateKafkaConfig(ctx context.Context, d *schema.ResourceData, client *godo.Client) error {
+	clusterID := d.Get("cluster_id").(string)
+
+	opts := &godo.KafkaConfig{}
+
+	if v, ok := d.GetOk("group_initial_rebalance_delay_ms"); ok {
+		opts.GroupInitialRebalanceDelayMs = godo.PtrTo(v.(int))
+	}
+
+	if v, ok := d.GetOk("group_min_session_timeout_ms"); ok {
+		opts.GroupMinSessionTimeoutMs = godo.PtrTo(v.(int))
+	}
+
+	if v, ok := d.GetOk("group_max_session_timeout_ms"); ok {
+		opts.GroupMaxSessionTimeoutMs = godo.PtrTo(v.(int))
+	}
+
+	if v, ok := d.GetOk("message_max_bytes"); ok {
+		opts.MessageMaxBytes = godo.PtrTo(v.(int))
+	}
+
+	if v, ok := d.GetOk("log_cleaner_delete_retention_ms"); ok {
+		opts.LogCleanerDeleteRetentionMs = godo.PtrTo(int64(v.(int)))
+	}
+
+	if v, ok := d.GetOk("log_cleaner_min_compaction_lag_ms"); ok {
+		uintVal, _ := strconv.ParseUint(v.(string), 10, 64)
+		opts.LogCleanerMinCompactionLagMs = godo.PtrTo(uintVal)
+	}
+
+	if v, ok := d.GetOk("log_flush_interval_ms"); ok {
+		uintVal, _ := strconv.ParseUint(v.(string), 10, 64)
```

Comment:
Can we check for errors here and other places we are using `strconv.ParseUint`? If a user puts something wrong there, we'll set it to `0` rather than erroring.

Reply:
Thanks :+1: That's a good idea
What would you say about this implementation: [88eedf8](https://github.com/digitalocean/terraform-provider-digitalocean/pull/1218/commits/88eedf81404bbcf60196d73d1587cbcf08c6882e)
We have something similar in https://github.com/digitalocean/terraform-provider-digitalocean/blob/88eedf81404bbcf60196d73d1587cbcf08c6882e/digitalocean/database/resource_database_cluster.go#L327